spectral_cube/spectral_cube.py

```diff
@@ -1856,7 +1898,49 @@ def subcube(self, xlo='min', xhi='max', ylo='min', yhi='max', zlo='min',
         # list to track which entries had units
         united = []
 
+        # Solve the spatial axes together. i.e., lower and upper corners
+        # as distance from (xlo, ylo) or (xhi, yhi)
+
+        for corn in ['lo', 'hi']:
+            grids = {}
+
+            x_hasunit = hasattr(limit_dict['x'+corn], 'unit')
+            y_hasunit = hasattr(limit_dict['y'+corn], 'unit')
+
+            if not any([x_hasunit, y_hasunit]):
+                continue
+
+            for lim in ['x', 'y']:
+
+                dim = 1 if lim == 'y' else 2
+
+                # WCS grid
+                grids[lim] = self.spatial_coordinate_map[dim-1]
+
+                united.append(lim + corn)
+
+            log.debug(f"Grid: {grids['x'][0, :]}, Corner: {limit_dict['x' + corn]}")
+            log.debug(f"Grid: {grids['y'][:, 0]}, Corner: {limit_dict['y' + corn]}")
+
+            x2 = (grids['x'] - limit_dict['x' + corn])**2
+            if hasattr(x2, 'unit'):
+                x2 = x2.value
+            y2 = (grids['y'] - limit_dict['y' + corn])**2
+            if hasattr(y2, 'unit'):
+                y2 = y2.value
+
+            dist = np.sqrt(x2 + y2)
+            # Get the min posns
+            ymin, xmin = np.unravel_index(dist.argmin(), dist.shape)
```

Comment:
Some of this duplicates some of the code higher up - I wonder if we can somehow factor some of it out into a helper function?

Reply:
I split it out into [wcs_utils.find_spatial_pixel_index](https://github.com/radio-astro-tools/spectral-cube/pull/666/files#diff-fc85cee04a934943754a8ab34f23bf74R486).